### PR TITLE
chore(deps): bump date-fns to v3

### PR DESCRIPTION
**Pull request by frsimond**
I try here to solve #655 and support last version of date-fns with its breaking changes.
I decided heer to keep a specific package for backwards compatibility. Not a fan of the resulting code duplication but it's a proposal.

### CODE REVIEW

**Comment by dmtrKovalenko** [comment]
I'm ok with duplication but we should probably bump the major version as well. Not sure for all of packages or not

**Comment by frsimond** [comment]
Ok. I totally agree that `@date-io/date-fns` should have a major version bump. I don't think we need to bump the version globally.
Should I bump the version(s) myself in this PR ?

**Comment by dmtrKovalenko** [comment]
We can do that later, but what is requried right now it to ensure that the tests are running for both versions of date-{ns

**Comment by frsimond** [comment]
Working on the tests, I decided to revert the date-fns-v2 because those have explicit reference to the date-fns package.
And I dont know how to reference two different versions of the same package in the same file.

Furthermore, I think that keeping a old version might be overengineered if we do a major version bump on `@date-io/date-fns`

**Comment by frsimond** [comment]
@dmtrKovalenko do you think we should maintain a compatibility with date-fns v2 on master ?
If so, we might have to change the structure of the repo a little bit to have package-specific tests files.

File: __tests__/formats.test.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ describe("Localized formats", () => {
   test.each`
     format                   | expectedWithEn                    | expectedWithRu
     ${"fullDate"}            | ${"Feb 1, 2020"}                  | ${"1 фев. 2020 г."}
-    ${"fullDateWithWeekday"} | ${"Saturday, February 1st, 2020"} | ${"суббота, 1-е февраля 2020 г."}
+    ${"fullDateWithWeekday"} | ${"Saturday, February 1st, 2020"} | ${"суббота, 1 февраля 2020 г."}
```
[the remark this answers is not in the document]
Comment (frsimond) [reply __tests__/formats.test.ts:88]: @dmtrKovalenko ?

**Comment by dmtrKovalenko** [comment]
Thank you for contributation, I am going to merge this right and now and make another breaking changes that I wanted to make for a long time and will release a new version today

**Pull request merged**